include/onnxruntime/core/framework/provider_options.h

```diff
@@ -15,4 +15,18 @@ using ProviderOptions = std::unordered_map<std::string, std::string>;
 using ProviderOptionsVector = std::vector<ProviderOptions>;
 using ProviderOptionsMap = std::unordered_map<std::string, ProviderOptions>;
 
+struct ProviderOptionsUtils {
+  // return the prefix that is used for provider options when they're in SessionOptions.config_options.
+  // the prefix is not used when ProviderOptions are coming from the user.
+  static std::string GetProviderOptionPrefix(const std::string& name) {
+    return GetProviderOptionPrefix(name.c_str());
+  }
+
+  // return the prefix that is used for provider options when they're in SessionOptions.config_options.
+  // the prefix is not used when ProviderOptions are coming from the user.
+  static std::string GetProviderOptionPrefix(const char* name) {
+    return std::string("ep.") + name + ".";
```

Comment:
Should this be made lowercase?

Reply:
yeah. we want it to be lowercase everywhere for consistency right? i.e. lowercase before prefixing when adding to SessionOptions.config_options.